packages/hub/src/hub.ts

```diff
@@ -381,17 +387,18 @@ export class Hub implements HubInterface {
   /**
    * @inheritDoc
    */
-  public endSession(): void {
+  public endSession(clearSessionFromScope: boolean = true): void {
```

Comment:
Rather than controlling the remove-from-scope question via a boolean, what would you think of having two separate methods instead? That way, every call to `endSession(false)` (which is pretty misleading, since it _doesn't_ actually end the session) would get to be a `captureSessionUpdate()` instead (which is actually what it's doing).

```
  public captureSessionUpdate(): void {
    const { scope, client } = this.getStackTop();
    if (!scope) return;

    const session = scope.getSession && scope.getSession();
    if (session) {
      if (client && client.captureSession) {
        client.captureSession(session);
      }
    }
  }
  
  public endSession(): void {
    captureSessionUpdate()
    
    // the session is over; take it off of the scope
    scope.setSession();
  }
```

If it were ever a runtime question whether or not to end a session but still have `endSession()` available on its own, you could change the signature to `captureSessionUpdate(endSession: boolean = false)` and pull the sending into a different method:

_(scrolling necessary)_

```
  private _sendSessionUpdate(): void {
    const { scope, client } = this.getStackTop();
    if (!scope) return;

    const session = scope.getSession && scope.getSession();
    if (session) {
      if (client && client.captureSession) {
        client.captureSession(session);
      }
    }
  }
  
  public captureSessionUpdate(endSession: boolean = true): void {
      // both send the update and pull the session from the scope
      if (endSession) {
        return this.endSession();
      }
      
      // only send the update
      this._sendSessionUpdate()
  }
  
  public endSession(): void {
    this._sendSessionUpdate()
    
    // the session is over; take it off of the scope
    scope.setSession();
  }
```

Reply:
Good refactor 👍